### ISSUE TITLE
Redesign Training Data Loading in Latent Rollout for Performance and Code Clarity

### ISSUE THREAD

**Issue by mlaurako** [NOAA-EMC/ocelot#53]
## Problem Statement

The current rollout training implementation loads data on-the-fly during training steps, creating performance bottlenecks:

- `_get_sequential_step_data()` calls `extract_features()` and `obs_mesh_conn()` **during training steps**
- Creates temporary `BinDataset` objects during GPU training time
- Wastes compute resources on data processing when the GPU should be focused on training

## Proposed Solution

Implement a `RolloutDataset` that pre-organizes rollout sequences during dataset setup:

**Current:**
```python
# During training step:
target_bin_data = data_module.data_summary[target_bin_name]  
temp_graph_data = data_module._create_graph_structure(target_bin_data)  # SLOW!
```

**Proposed:**
```python
# During dataset setup:
rollout_sequences = [(bin7, bin8, bin9), (bin8, bin9, bin10), ...]  # Pre-organized
# During training: just use pre-loaded graphs
```

## Benefits

1. **Performance**: Eliminates data processing during training steps
2. **Code Readability**: Cleaner separation between data loading and training logic  
3. **Preserves Existing Code**: Core data processing utilities (`extract_features`, `obs_mesh_conn`, `organize_bins_times`) remain unchanged
4. **Better Architecture**: Removes the complex `_get_sequential_step_data` method
5. **Easier Debugging**: Pre-loaded sequences are easier to inspect and validate

## Implementation Plan

- [ ] Create `RolloutDataset` class for pre-organizing rollout sequences
- [ ] Update `GNNDataModule` to use `RolloutDataset` for training  
- [ ] Modify `GNNLightning.training_step()` to handle pre-loaded sequences
- [ ] Remove `_get_sequential_step_data()` method
- [ ] Add backward compatibility flag for single-bin training
- [ ] Update training script with rollout configuration options

## Scope

This redesign focuses on **how** rollout data is loaded and organized, not **what** data is processed. All existing data processing logic, mesh creation, and model architecture remain unchanged.

## Expected Outcome

Faster rollout training with cleaner, more maintainable code that separates data loading concerns from training logic.

**Comment by mlaurako** [comment]
## Update data boundary handling
Add rollout buffer management to prevent data loss at window boundaries. For example, with 3-step rollout on a 14-day window, day 14 cannot form a complete rollout sequence (needs days 15-16), so we can extend the data loading by 1-2 additional days to create a rollout buffer, while ensuring this buffer doesn't overlap with the validation set.
